src/PlatformJson.js

```diff
@@ -211,8 +210,7 @@ PlatformJson.prototype.generateMetadata = function () {
  */
 PlatformJson.prototype.generateAndSaveMetadata = function (destination) {
     var meta = this.generateMetadata();
-    fs.ensureDirSync(path.dirname(destination));
-    fs.writeFileSync(destination, meta, 'utf-8');
+    fs.outputJsonSync(destination, meta);
```

Comment:
I hate to bug you again with this, but it was not about re-spacing JSON output. `generateMetadata` returns a string containing JS, **not JSON**. Therefore we need `fs.outputFileSync(destination, meta)` here.

I also wrote a butt-ugly test that will fail with your implementation:
```js
describe('generateAndSaveMetadata method', function () {
    it('should save generated metadata', function () {
        // Needs to use graceful-fs, since that is used by fs-extra
        const spy = spyOn(require('graceful-fs'), 'writeFileSync');

        const dest = require('path').join(__dirname, 'test-destination');
        platformJson.addPluginMetadata(fakePlugin).generateAndSaveMetadata(dest);

        expect(spy).toHaveBeenCalledTimes(1);
        const [file, data] = spy.calls.argsFor(0);
        expect(file).toBe(dest);
        expect(data.indexOf(JSON.stringify(platformJson.root.modules, null, 2))).toBeGreaterThan(0);
        expect(data).toMatch(JSON.stringify(platformJson.root.plugin_metadata, null, 2));
    });
});
```

Reply:
Ahh, now I understand! :man_facepalming: 

I've fixed that and added your test to the PlatformJson.spec.js